ecc_enclave/sgxcclib/CMakeVariables.txt

```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SET(TCC_ATTESTATION_TYPE "$ENV{TCC_ATTESTATION_TYPE}")
```

Comment:
Maybe this might be better called TCC_EPID_LINKAGE_TYPE?  I would expect eventually we will have TCC_ATTESTATION_TYPE in { EPID, DCAP, MOCK } ?

Reply:
I suggest to call it `ECC_*` is it currently resides in `ecc_enclave`. I also prefer `ECC_EPID_LINKAGE_TYPE`

Reply:
Let me ask for one more iteration considering that:
-"epid linkage" has never been used before
-the variable is related to the value that will go inside an sgx_quote_sign_type_t structure
-it is not about ECC, it is about the credentials owner by the peer (?) (perhaps PEER, or nothing, is better?)

Good news: we agree on the "..._TYPE" :D

Reply:
I guess there are two parts of the naming discussion:
- what is the prefix: should it be a particular component (e.g., ERCC) or should it be the overall project (although in that case it really should be SCC or PCC from our past discussions, not TCC)
- if we restrict to linkable vs unlinkable whether "attestation" is not too generic given that we will have other attestation types besides EPID where linkable vs unlinkable is not really meaningful.  I would see two ways to resolve this: either (a) we keep the environment variable generic as *_ATTESTATION_TYPE' but make the value specific to EPID_LINKABLE (vs EPID_UNLINKABLE vs MOCK vs DCAP ...) or (b) we make this environment variable specific to EPID linkage ...

Reply:
> what is the prefix: should it be a particular component (e.g., ERCC) or should it be the overall project (although in that case it really should be SCC or PCC from our past discussions, not TCC)

agree, I vote for component prefix as one may use different attestations types for different components in the future? Also, as the naming is not completely resolved yet, let's just call it `ECC_`. Then it's clear that we refer to the attestation for a specific component and matches the current naming scheme.

> (a) we keep the environment variable generic as *_ATTESTATION_TYPE

I've changed my mind. I vote for
```suggestion
ECC_ATTESTATION_TYPE = {EPID_LINKABLE|EPID_UNLINKABLE|MOCK|DCAP}
```

This looks pretty clean to me.

Reply:
alright:
-ECC_ATTESTATION_TYPE good for everybody
-string must be epid_linkable or epid_unlinkable (lower case, to differentiate it from upper case variables)
-C compile time defines are USE_EPID_(UN)LINKABLE
-C code define is SGX_QUOTE_SIGN_TYPE, to maintain consistency with the data structure it refers to